### ISSUE TITLE
Player error after changing parameters of player's track selector - Attempt to read from null array

### ISSUE THREAD

**Issue by hsahnv** [google/ExoPlayer#6746]
### [REQUIRED] Issue description
Player error occurs when trying to change parameters of player's track selector.
(About 80% error)

**Parameter changing codes:**
```
public class CustomTrackSelector extends DefaultTrackSelector {
    // ...
    public void clearVideoSizeConstraints() {
        ParametersBuilder builder = getParameters().buildUpon();
        builder.clearVideoSizeConstraints();
        setParameters(builder.build());
    }
    // ...
}
```

**Player error:**

```
playerFailed [0.69, 3.78, window=0, period=0]
    com.google.android.exoplayer2.ExoPlaybackException: java.lang.NullPointerException: Attempt to read from null array
        at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:397)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:193)
        at android.os.HandlerThread.run(HandlerThread.java:65)
     Caused by: java.lang.NullPointerException: Attempt to read from null array
        at com.google.android.exoplayer2.source.hls.HlsSampleStreamWrapper.selectTracks(HlsSampleStreamWrapper.java:329)
        at com.google.android.exoplayer2.source.hls.HlsMediaPeriod.selectTracks(HlsMediaPeriod.java:284)
        at com.google.android.exoplayer2.MediaPeriodHolder.applyTrackSelection(MediaPeriodHolder.java:279)
        at com.google.android.exoplayer2.ExoPlayerImplInternal.reselectTracksInternal(ExoPlayerImplInternal.java:1126)
        at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:357)
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.os.HandlerThread.run(HandlerThread.java:65) 
```

### [REQUIRED] Reproduction steps
Change parameters of player's track selector immediately after initializing player.
**(I temporarily delay(500ms) parameter changing codes to avoid this issue and it's working.)**
See below logs.

### [REQUIRED] Link to test content
None

### [REQUIRED] A full bug report captured from the device

```
2019-12-11 11:25:09.960 7818-7818 D/EventLogger: state [0.00, 0.00, window=0, true, IDLE]
2019-12-11 11:25:09.964 7818-7818 D/EventLogger: seekStarted [0.01, 0.00, window=0]
2019-12-11 11:25:09.964 7818-7818 D/EventLogger: positionDiscontinuity [0.01, 3.78, window=0, SEEK]
2019-12-11 11:25:09.965 7818-7818 D/EventLogger: state [0.01, 3.78, window=0, true, BUFFERING]
2019-12-11 11:25:10.016 7818-7818 D/EventLogger: timelineChanged [0.06, 3.78, window=0, periodCount=1, windowCount=1, reason=PREPARED
2019-12-11 11:25:10.016 7818-7818 D/EventLogger:   period [182.45]
2019-12-11 11:25:10.016 7818-7818 D/EventLogger:   window [182.45, true, false]
2019-12-11 11:25:10.016 7818-7818 D/EventLogger: ]
2019-12-11 11:25:10.018 7818-7818 D/EventLogger: seekProcessed [0.06, 3.78, window=0]
2019-12-11 11:25:10.018 7818-7818 D/EventLogger: mediaPeriodCreated [0.06, 3.78, window=0, period=0]
2019-12-11 11:25:10.034 7818-7818 D/EventLogger: loading [0.08, 3.78, window=0, period=0, true]
2019-12-11 11:25:10.039 7818-7818 D/EventLogger: decoderEnabled [0.08, 3.78, window=0, period=0, video]
2019-12-11 11:25:10.039 7818-7818 D/EventLogger: decoderEnabled [0.08, 3.78, window=0, period=0, audio]
2019-12-11 11:25:10.040 7818-7818 D/EventLogger: tracksChanged [0.08, 3.78, window=0, period=0, 
2019-12-11 11:25:10.040 7818-7818 D/EventLogger:   Renderer:0 [
2019-12-11 11:25:10.040 7818-7818 D/EventLogger:     Group:0, adaptive_supported=YES [
2019-12-11 11:25:10.040 7818-7818 D/EventLogger:       [ ] Track:0, id=0, mimeType=video/avc, bitrate=10391667, codecs=avc1.4d4028, res=1920x1080, fps=29.97, supported=YES
2019-12-11 11:25:10.040 7818-7818 D/EventLogger:       [X] Track:1, id=1, mimeType=video/avc, bitrate=3418348, codecs=avc1.4d401f, res=1280x720, fps=29.97, supported=YES
2019-12-11 11:25:10.040 7818-7818 D/EventLogger:       [X] Track:2, id=2, mimeType=video/avc, bitrate=1675259, codecs=avc1.4d401f, res=854x480, fps=29.97, supported=YES
2019-12-11 11:25:10.040 7818-7818 D/EventLogger:     ]
2019-12-11 11:25:10.040 7818-7818 D/EventLogger:     Metadata [
2019-12-11 11:25:10.040 7818-7818 D/EventLogger:       com.google.android.exoplayer2.source.hls.HlsTrackMetadataEntry@d690f38
2019-12-11 11:25:10.040 7818-7818 D/EventLogger:     ]
2019-12-11 11:25:10.040 7818-7818 D/EventLogger:   ]
2019-12-11 11:25:10.040 7818-7818 D/EventLogger:   Renderer:1 [
2019-12-11 11:25:10.040 7818-7818 D/EventLogger:     Group:0, adaptive_supported=N/A [
2019-12-11 11:25:10.040 7818-7818 D/EventLogger:       [X] Track:0, id=0, mimeType=audio/mp4a-latm, codecs=mp4a.40.2, supported=YES
2019-12-11 11:25:10.040 7818-7818 D/EventLogger:     ]
2019-12-11 11:25:10.040 7818-7818 D/EventLogger:     Metadata [
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:       com.google.android.exoplayer2.source.hls.HlsTrackMetadataEntry@7bb65b3f
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:     ]
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:   ]
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:   Renderer:2 [
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:     Group:0, adaptive_supported=N/A [
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:       [ ] Track:0, id=subs:English, mimeType=text/vtt, language=en, label=English, supported=YES
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:     ]
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:     Group:1, adaptive_supported=N/A [
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:       [ ] Track:0, id=subs:Spanish, mimeType=text/vtt, language=es, label=Spanish, supported=YES
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:     ]
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:     Group:2, adaptive_supported=N/A [
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:       [X] Track:0, id=subs:Korean, mimeType=text/vtt, language=ko, label=Korean, supported=YES
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:     ]
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:     Group:3, adaptive_supported=N/A [
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:       [ ] Track:0, id=subs:Portuguese, mimeType=text/vtt, language=pt, label=Portuguese, supported=YES
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:     ]
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:     Group:4, adaptive_supported=N/A [
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:       [ ] Track:0, id=subs:Russian, mimeType=text/vtt, language=ru, label=Russian, supported=YES
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:     ]
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:     Metadata [
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:       com.google.android.exoplayer2.source.hls.HlsTrackMetadataEntry@e2f6cd0
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:     ]
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:   ]
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:   Renderer:3 [
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:     Group:0, adaptive_supported=N/A [
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:       [X] Track:0, id=ID3, mimeType=application/id3, supported=YES
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:     ]
2019-12-11 11:25:10.041 7818-7818 D/EventLogger:   ]
2019-12-11 11:25:10.041 7818-7818 D/EventLogger: ]
2019-12-11 11:25:10.045 7818-7818 D/EventLogger: mediaPeriodReadingStarted [0.09, 3.78, window=0, period=0]
2019-12-11 11:25:10.647 7818-7818 E/EventLogger: playerFailed [0.69, 3.78, window=0, period=0]
    com.google.android.exoplayer2.ExoPlaybackException: java.lang.NullPointerException: Attempt to read from null array
        at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:397)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:193)
        at android.os.HandlerThread.run(HandlerThread.java:65)
     Caused by: java.lang.NullPointerException: Attempt to read from null array
        at com.google.android.exoplayer2.source.hls.HlsSampleStreamWrapper.selectTracks(HlsSampleStreamWrapper.java:329)
        at com.google.android.exoplayer2.source.hls.HlsMediaPeriod.selectTracks(HlsMediaPeriod.java:284)
        at com.google.android.exoplayer2.MediaPeriodHolder.applyTrackSelection(MediaPeriodHolder.java:279)
        at com.google.android.exoplayer2.ExoPlayerImplInternal.reselectTracksInternal(ExoPlayerImplInternal.java:1126)
        at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:357)
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.os.HandlerThread.run(HandlerThread.java:65) 
2019-12-11 11:25:10.654 7818-7818 D/EventLogger: decoderDisabled [0.70, 3.78, window=0, period=0, video]
2019-12-11 11:25:10.654 7818-7818 D/EventLogger: decoderDisabled [0.70, 3.78, window=0, period=0, audio]
2019-12-11 11:25:10.654 7818-7818 D/EventLogger: mediaPeriodReleased [0.70, 3.78, window=0, period=0]
2019-12-11 11:25:10.655 7818-7818 D/EventLogger: loading [0.70, 3.78, window=0, false]
2019-12-11 11:25:10.655 7818-7818 D/EventLogger: state [0.70, 3.78, window=0, true, IDLE]
```

### [REQUIRED] Version of ExoPlayer being used
Exoplayer 2.10.8

### [REQUIRED] Device(s) and version(s) of Android being used
Pixel / Android SDK 28
Galaxy S8+ (SM-G955N) / Android SDK 28

**Comment by AquilesCanta** [comment]
Hi, I am not able to reproduce the issue. Please provide precise steps to reproduce the issue. Please:
- Include the stream which you are using to reproduce the issue.
- Provide the precise minimum sequence of steps you can take in the demo app to reproduce the issue.
- Frame the provided steps in the dev branch.

Things I've tried:
Added 

```java
  public void clearVideoSizeConstraints() {
    ParametersBuilder builder = getParameters().buildUpon();
    builder.clearVideoSizeConstraints();
    setParameters(builder.build());
  }
```
to DefaultTrackSelector. In the demo app's player activity, I am configuring the track selector as follows:

```java
trackSelector = new DefaultTrackSelector(trackSelectionFactory);
trackSelector.setParameters(
    trackSelectorParameters.buildUpon().setMaxVideoSize(10, 10).build());
```

(I have also tried without this change) And then, after 

`player.prepare(mediaSource, !haveStartPosition, false);`

in the PlayerActivity of the demo app, I am calling `trackSelector.clearVideoSizeConstraints()`. I have also tried `new Handler().post(() -> trackSelector.clearVideoSizeConstraints());`.

I have tried these things using the HLS streams available in the demo app.

**Comment by AquilesCanta** [comment]
I suspect this may be fixed in the dev-branch. Please confirm.

EDIT: I haven't been able to reproduce, but I think I know which code path could cause this. Are you using chunkless preparation?

**Comment by hsahnv** [comment]
@AquilesCanta Thanks for your comments.
In this case, because of the app-dependent code, it seems difficult to reproduce this problem in a demo app.

I'm using chunkless preparation. And this issue disappeared after disallowing chunkless preparaion.
I will test again with dev-branch version exoplayer and let you know the result.

**Comment by hsahnv** [comment]
It seems to be fixed in the dev-branch.
I tested dev-v2 (5bbabbc14de07ab40e337ba37f6622a3ec59950f) and this issue isn't reproduced anymore. ->  I have found the commit included in 2.11.0 and released.
Thanks.
